File: be/src/models/CommunityRule.js
const mongoose = require('mongoose');

const communityRuleSchema = new mongoose.Schema(
  {
    code: {
      type: String,
      required: true,
      unique: true,
      trim: true,
      maxlength: 60,
      index: true,
    },
    title: {
      type: String,
      required: true,
      trim: true,
      maxlength: 120,
    },
    description: {
      type: String,
      required: true,
      maxlength: 1000,
    },
    category: {
      type: String,
      enum: ['SAFETY', 'RESPECT', 'PRIVACY', 'CONTENT', 'OTHER'],
      default: 'OTHER',
      index: true,
    },
    severity: {
      type: String,
      enum: ['LOW', 'MEDIUM', 'HIGH'],
      default: 'MEDIUM',
      index: true,
    },
    isActive: {
      type: Boolean,
      default: true,
      index: true,
    },
  },
  { timestamps: true },
);

module.exports = mongoose.model('CommunityRule', communityRuleSchema);

File: fe/lib/core/models/community_rule.dart
class CommunityRule {
  const CommunityRule({
    required this.id,
    required this.title,
    required this.description,
  });

  final String id;
  final String title;
  final String description;

  factory CommunityRule.fromJson(Map<String, dynamic> json) {
    return CommunityRule(
      id: (json['id'] ?? '').toString(),
      title: (json['title'] ?? '').toString(),
      description: (json['description'] ?? '').toString(),
    );
  }
}
